src/computation/logic/distribute-levelling-seats.ts

```diff
@@ -21,75 +100,124 @@ export function distributeLevelingSeats(
+) {
+    let totalVotes = 0;
+    const levelingParties = [];
+    for (const partyCode of levelingPartyCodes) {
+        totalVotes += partyResults[partyCode].votes;
+        const party: Result = {
+            countyId: -1,
+            electionId: -1,
+            partyId: -1,
+            resultId: -1,
+            countyName: "",
+            partyCode,
+            partyName: "",
+            votes: partyResults[partyCode].votes,
+            percentage: partyResults[partyCode].percentVotes,
+        };
+        levelingParties.push(party);
+    }
 
     // Distribute the leveling seats, taking the district seats into account
-    const levelingSeatsDistribution = distributeSeats(
+    return distributeSeats(
         payload.algorithm,
         payload.firstDivisor,
         Number.MIN_SAFE_INTEGER,
         payload.levelingSeats,
+        totalVotes,
         levelingParties,
         undefined,
         partyResults
     );
+}
 
-    for (const partyCode in levelingSeatsDistribution.seatsWon) {
-        if (levelingSeatsDistribution.seatsWon.hasOwnProperty(partyCode)) {
-            const levelingSeats = levelingSeatsDistribution.seatsWon[partyCode];
-            partyResults[partyCode].levelingSeats += levelingSeats;
-            partyResults[partyCode].totalSeats += levelingSeats;
-        }
-    }
-
-    levelingPartyCodes = levelingPartyCodes.filter((partyCode) => partyResults[partyCode].levelingSeats > 0);
-
-    let partyRestQuotients: Dictionary<PartyRestQuotients> = {};
-
-    if (payload.election.year < 2005) {
-        partyRestQuotients = distributeLevelingSeatsOnDistrictsPre2005(
-            payload,
-            levelingPartyCodes,
-            partyResults,
-            districtPartyResults,
-            districtResults
-        );
-    } else {
-        partyRestQuotients = distributeLevelingSeatsOnDistricts(
-            payload,
-            levelingPartyCodes,
-            partyResults,
-            districtPartyResults,
-            districtResults
-        );
-    }
+/**
+ * Distributes the leveling seats on each party using the largest fraction algorithm.
+ *
+ * @param levelingPartyCodes The party codes for the parties that may win a leveling seat
+ * @param partyResults The party results for each party
+ * @param nationalDistribution The final national distribution of seats
+ */
+function finalLargestFractionLevelingSeatDistribution(
+    levelingPartyCodes: string[],
+    partyResults: Dictionary<PartyResult>,
+    nationalDistribution: DistributionResult
+): DistributionResult {
+    const levelingSeatDistribution: Dictionary<number> = {};
+    levelingPartyCodes.forEach((partyCode) => {
+        levelingSeatDistribution[partyCode] =
+            nationalDistribution.seatsWon[partyCode] - partyResults[partyCode].districtSeats;
+    });
 
-    const levelingSeatDistribution = dictionaryToArray(partyRestQuotients);
-    return levelingSeatDistribution;
+    return {
+        seatResults: [],
```

Comment:
Why is an empty array being returned here?

Reply:
Because levelling seats are not distributed in rounds like Sainte Laguës or d'Hondt. So there is not necessarily any definition of who won a round, as all seats may be distributed in a single round.